cassandane/Cassandane/Cyrus/Sieve.pm

```diff
@@ -7133,4 +7133,28 @@ EOF
     $self->check_messages({ 1 => $msg }, check_guid => 0);
 }
 
+sub test_redirect_address_with_phrase
+    :needs_component_sieve
+{
+    my ($self) = @_;
+
+    $self->{store}->set_fetch_attributes(qw(uid flags));
+
+    xlog $self, "Install a script";
+    $self->{instance}->install_sieve_script(<<EOF
+redirect "Foo <foo\@example.com>";
+EOF
+    );
+
+    xlog $self, "Deliver a message";
+    my $msg = $self->{gen}->generate(subject => "Message 1");
+    $self->{instance}->deliver($msg);
+
+    # Verify that message was sent
+    if ($self->{instance}->{have_syslog_replacement}) {
+        my @lines = $self->{instance}->getsyslog();
+        $self->assert_does_not_match(qr/RCPT TO: code=553 text=5.1.1/, "@lines");
+    }
```

Comment:
For a belt-and-suspenders approach, you could also use `check_messages()` to make sure the message actually arrived.  That way, it would keep the test valid even when run in environments where cassandane can't capture syslog.  I guess you'd need to add a "foo@example.com" user to the instance for that to work... there's examples around of that sort of thing, have a look at `test_plus_address_partial_virtdom()` in `Cassandane::Cyrus::Delivery` for one.

Reply:
@elliefm I don't think SMTPServer.pm actually does message delivery, so there would be no redirected message to look for.

However, you suggestion made me think: If the redirect is successful, the original recipient would NOT have the message delivered to their INBOX (unless :copy is used).  It would only end up in their INBOX if the script encounters a runtime error (from the redirect action).  So, I added a check for that.

Reply:
Ahh yeah nice!